### PR TITLE
Split/Refacto main module

### PR DESCRIPTION
**Pull request by o2sh**
Did a bit of cleaning/refactoring. Starting with the `main.rs` module. I extracted the `onefetch -h` part into `cli.rs`. I also created an `Option` struct that holds the input config and pass it along to `Info.rs` 's constructor.

Next step will be to refactor `Info.rs`...hopefully before it breaks the 1000 line barrier :sunglasses:

### CODE REVIEW

File: src/onefetch/license.rs [hunk 1]
```diff
@@ -1,10 +1,10 @@
 use askalono::{Store, TextData};
 
-use crate::Error;
+use crate::onefetch::error::*;
 
-type Result<T> = std::result::Result<T, Error>;
+pub const LICENSE_FILES: [&str; 3] = ["LICENSE", "LICENCE", "COPYING"];
 
-static CACHE_DATA: &[u8] = include_bytes!("../resources/licenses/cache.bin.zstd");
+static CACHE_DATA: &[u8] = include_bytes!("../../resources/licenses/cache.bin.zstd");
```
Comment (spenserblack) [inline src/onefetch/license.rs:7]:
```suggestion
static CACHE_DATA: &[u8] = include_bytes!(concat!(env!("CARGO_MANIFEST_DIR"), "/resources/licenses/cache.bin.zstd"));
```

**Review by spenserblack — APPROVED** [review]
LGTM :+1:
Just some small notes/suggestions.

File: src/onefetch/cli.rs [hunk 2]
```diff
@@ -0,0 +1,200 @@
...
+    pub ascii_colors: Vec<String>,
+    pub disabled_fields: info_fields::InfoFieldOn,
+    pub no_bold: bool,
+    pub image: Option<DynamicImage>,
+    pub image_backend: Option<Box<dyn image_backends::ImageBackend>>,
+    pub no_merges: bool,
+    pub no_color_blocks: bool,
+    pub number_of_authors: usize,
+    pub excluded: Vec<String>,
+}
```
[the remark this answers is not in the document]
Comment (o2sh) [reply src/onefetch/cli.rs:23]: I'm not sure this syntax will be able to cover our needs (at least not in one go). Especially for the option's parameters that require post-processing like `image-backend` and `disabled_fields`. I could however use this syntax on a simplified `Opt` struct and then map it to the real `Option` struct.

[the remark this answers is not in the document]
Comment (spenserblack) [reply src/onefetch/cli.rs:23]: Yeah, not saying using structopt is something to do for this PR. Just wanted to bring up the option before I forget it :laughing:. We may want to use it eventually just to remove the manual definition of all the clap args that then get mapped to this struct.

>I could however use this syntax on a simplified `Opt` struct and then map it to the real `Option` struct.

Also, the `structopt::StructOpt` trait can be implemented on a wrapper struct for these fields.

[the remark this answers is not in the document]
Comment (o2sh) [reply src/onefetch/cli.rs:23]: > Also, the structopt::StructOpt trait can be implemented on a wrapper struct for these fields.

Great Idea, I may try it on another PR.

**Pull request merged**